cmd/compose/compose.go

```diff
@@ -354,10 +354,8 @@ func setEnvWithDotEnv(prjOpts *projectOptions) error {
 		return err
 	}
 	for k, v := range envFromFile {
-		if _, ok := os.LookupEnv(k); !ok {
-			if err := os.Setenv(k, v); err != nil {
-				return err
-			}
+		if err := os.Setenv(k, v); err != nil {
+			return err
 		}
 	}
```

Comment:
Nit: Given the complexity/subtlety, I think a comment here is warranted, e.g. `// overwrite the process env with merged OS + env file results` or something

For context: I thought this was a regression at first since OS/shell has higher precedence than env file, but then I realized that `GetEnvFromFile` takes in the OS/shell env and handles the precedence/merging, which is why it's safe/necessary to overwrite everything here

Reply:
Yep! Adding the comment.